src/taipy/core/scenario/_scenario_manager_factory.py

```diff
@@ -13,16 +13,21 @@
 
 from .._manager._manager_factory import _ManagerFactory
 from ..common._utils import _load_fct
-from ._scenario_manager import _ScenarioManager
-from ._scenario_repository_factory import _ScenarioRepositoryFactory
+from ._scenario_manager import _build_repository, _ScenarioManager
 
 
 class _ScenarioManagerFactory(_ManagerFactory):
     @classmethod
     def _build_manager(cls) -> Type[_ScenarioManager]:  # type: ignore
         if cls._using_enterprise():
-            return _load_fct(
+            scenario_manager = _load_fct(
                 cls._TAIPY_ENTERPRISE_CORE_MODULE + ".scenario._scenario_manager", "_ScenarioManager"
             )  # type: ignore
-        _ScenarioManager._repository = _ScenarioRepositoryFactory._build_repository()  # type: ignore
+            build_repository = _load_fct(
+                cls._TAIPY_ENTERPRISE_CORE_MODULE + ".scenario._scenario_manager", "_build_repository"
+            )  # type: ignore
+        else:
+            scenario_manager = _ScenarioManager
```

Comment:
I don't understand why you return the class `_ScenarioManager` and not an instance `_ScenarioManager()`.

BTW I got the same question for all manager factories

Reply:
because _ScenarioManager is a singleton (not quite so in terms of implementation but it was intended that way I believe). If you check the code of the ScenarioManager, there's no concrete function or self.attributes, only classmethods

Reply:
OK.

For me, the managers were not singletons but classes to instantiate that hold a specific repository.

But you are right. There is no state in the managers except the repository. And once the repository is set in the manager, it should not be modified. So it is kind of a singleton. That's fine for me.